backend/sirius-web-services-api/src/main/java/org/eclipse/sirius/web/services/api/images/ICustomImagesService.java

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.services.api.images;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Service layer for custom images.
+ *
+ * @author pcdavid
+ */
+public interface ICustomImagesService {
+    Optional<CustomImage> findById(UUID id);
+
+    Optional<byte[]> getImageContentsByFileName(String fileName);
+
+    List<CustomImage> getAvailableImages();
+
+    /**
+     * Null implementation.
+     *
+     * @author pcdavid
+     */
+    class NoOp implements ICustomImagesService {
+        @Override
+        public Optional<byte[]> getImageContentsByFileName(String fileName) {
+            return Optional.empty();
+        }
+
+        @Override
+        public List<CustomImage> getAvailableImages() {
+            return List.of();
+        }
+
+        @Override
+        public Optional<CustomImage> findById(UUID id) {
+            return Optional.empty();
+        }
+    }
```

Comment:
Why have an interface containing its implementation?

Reply:
It was kind of a test/proposal to avoid the current situation where we have to create multiple NoOp implementations of our APIs (e.g. 2 verions of `NoOpEditingContext`, 3 `NoOpEditService`, etc.).
Granted, in this particular case this is used at a single location, but it was more to see your reaction about this approach :smiley: 
I can use the current pattern instead and move `NoOpCustomImageService` in the single location it is used at the moment (close to `ImagesControllerTestCases`).